src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -83,97 +80,93 @@ protected CodeGenerator(InsertionPoint insertionPoint, SelectionResult selection
-
-                    newContainer = codeGenerationService.AddMethod(destination, result.Data, info, cancellationToken);
-                }
+                        documentWithUpdatedCallSite.Project.ParseOptions);
 
-                var newSyntaxRoot = newCallSiteRoot.ReplaceNode(destination, newContainer);
-                var newDocument = callSiteDocument.Document.WithSyntaxRoot(newSyntaxRoot);
-
-                var generatedDocument = await SemanticDocument.CreateAsync(newDocument, cancellationToken).ConfigureAwait(false);
-
-                // For nullable reference types, we can provide a better experience by reducing use 
-                // of nullable reference types after a method is done being generated. If we can
-                // determine that the method never returns null, for example, then we can
-                // make the signature into a non-null reference type even though
-                // the original type was nullable. This allows our code generation to
-                // follow our recommendation of only using nullable when necessary.
-                // This is done after method generation instead of at analyzer time because it's purely
-                // based on the resulting code, which the generator can modify as needed. If return statements
-                // are added, the flow analysis could change to indicate something different. It's cleaner to rely
-                // on flow analysis of the final resulting code than to try and predict from the analyzer what 
-                // will happen in the generator. 
-                var finalDocument = await UpdateMethodAfterGenerationAsync(generatedDocument, result, cancellationToken).ConfigureAwait(false);
-                var finalRoot = finalDocument.Root;
-
-                var methodDefinition = finalRoot.GetAnnotatedNodesAndTokens(MethodDefinitionAnnotation).FirstOrDefault();
-                if (!methodDefinition.IsNode || methodDefinition.AsNode() == null)
-                {
-                    return await CreateGeneratedCodeAsync(
-                        result.Status.With(OperationStatus.FailedWithUnknownReason), finalDocument, cancellationToken).ConfigureAwait(false);
+                    var newContainer = codeGenerationService.AddMethod(destination, newMethodDefinition, info, cancellationToken);
+                    var finalRoot = documentWithUpdatedCallSite.Root.ReplaceNode(destination, newContainer);
+                    return finalRoot;
                 }
-
-                if (methodDefinition.SyntaxTree.IsHiddenPosition(methodDefinition.AsNode().SpanStart, cancellationToken) ||
-                    methodDefinition.SyntaxTree.IsHiddenPosition(methodDefinition.AsNode().Span.End, cancellationToken))
-                {
-                    return await CreateGeneratedCodeAsync(
-                        result.Status.With(OperationStatus.OverlapsHiddenPosition), finalDocument, cancellationToken).ConfigureAwait(false);
-                }
```

Comment:
this logic moved up to the syntactic check in MethodExtractor (`CanAddTo`).  There we pre-check the destination tom ake sure there is a location to insert a new method into.  So this can't happen and no longer needs to be checked here.